ddtrace/tracer/context.go

```diff
@@ -46,6 +46,13 @@ func StartSpanFromContext(ctx context.Context, operationName string, opts ...Sta
 	if s, ok := SpanFromContext(ctx); ok {
 		opts = append(opts, ChildOf(s.Context()))
 	}
+	opts = append(opts, withContext(ctx))
 	s := StartSpan(operationName, opts...)
+	// If profiler labels were applied for this span, use the derived ctx that
+	// includes them. Otherwise a child of this span wouldn't be able to
+	// correctly restore the labels of its parent when it finishes.
+	if span, ok := s.(*span); ok && span.activeContext != nil {
```

Comment:
```suggestion
	if span, ok := s.(*span); ok && span.activeContext != nil {
		// If profiler labels were applied for this span, use the derived ctx that
		// includes them. Otherwise a child of this span wouldn't be able to
		// correctly restore the labels of its parent when it finishes.
```
Can we please do this? https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#comments